src/Tokenizer.js

```diff
@@ -370,32 +374,30 @@ module.exports = class Tokenizer {
           }
         }
 
-        l = item.cells.length;
+        l = item.cells.text.length;
         for (i = 0; i < l; i++) {
-          item.cells[i] = splitCells(item.cells[i], item.header.length);
+          item.cells.text[i] = splitCells(item.cells.text[i], item.header.text.length);
         }
 
         // parse child tokens inside headers and cells
-        item.tokens = {
-          header: [],
-          cells: []
-        };
+        item.header.tokens = [];
+        item.cells.tokens = [];
 
         // header child tokens
-        l = item.header.length;
+        l = item.header.text.length;
         for (j = 0; j < l; j++) {
-          item.tokens.header[j] = [];
-          this.lexer.inlineTokens(item.header[j], item.tokens.header[j]);
+          item.header.tokens[j] = [];
+          this.lexer.inlineTokens(item.header.text[j], item.header.tokens[j]);
```

Comment:
Now that I think about it the `tokens` property here is an array of token arrays.

Maybe instead we should do:

```js
{
  header: {
    items: [
      {
        text: 'header cell 1 text',
        tokens: [/* array of tokens for header cell 1 */]
      },
      {
        text: 'header cell 2 text',
        tokens: [/* array of tokens for header cell 2 */]
      }
    ]
  }
	cells: [
    { // row 1
      items: [
        {
          text: 'row 1 cell 1 text',
          tokens: [/* array of tokens for row 1 cell 1 */]
        },
        {
          text: 'row 1 cell 2 text',
          tokens: [/* array of tokens for row 1 cell 2 */]
        }
      ]
    },
    { // row 2
      items: [
        {
          text: 'row 2 cell 1 text',
          tokens: [/* array of tokens for row 2 cell 1 */]
        },
        {
          text: 'row 2 cell 2 text',
          tokens: [/* array of tokens for row 2 cell 2 */]
        }
      ]
    }
  ]
}
```

That seems to be more consistent with the rest of the tokens (including lists)

Reply:
Hm. Good point. I have a couple more thoughts:

First.... Is it worth it to have `header` and `row` as objects with a single property `items`? An object with a  single property seems redundant and maybe slightly slows things with extra object accesses? What if instead `header` and `rows` are just arrays like `items` is an array? So a List has `Items`, a Table has `Headers` and `Rows`, but they use the same format of being an array of "sub-tokens":

```
{
  type: 'table',
  align: [null, null],
  raw: '| a | b |\n|---|---|\n| 1 | 2 |\n',
  header: [
    {
      text: 'a',
      tokens: [{ type: 'text', raw: 'a', text: 'a' }],
    },
    {
      text: 'b',
      tokens: [{ type: 'text', raw: 'b', text: 'b' }],
    }
  ],
  rows: [ // replace `cells` with `rows`
    [ // row 1
      {
        text: '1',
        tokens: [{ type: 'text', raw: '1', text: '1' }],
      },
      {
        text: '2',
        tokens: [{ type: 'text', raw: '2', text: '2' }],
      }
    ]
  ]
}
```

Reply:
That sounds good

Reply:
@UziTech This is now changed. Want another look at it?